openedx/core/djangoapps/credit/signals.py

```diff
@@ -18,3 +21,42 @@ def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=
     from .tasks import update_credit_course_requirements
 
     update_credit_course_requirements.delay(unicode(course_key))
+
+
+@receiver(MIN_GRADE_REQUIREMENT_STATUS)
+def listen_for_grade_calculation(sender, username, grade_summary, course_key, deadline, **kwargs):  # pylint: disable=unused-argument
+    """Receive 'MIN_GRADE_REQUIREMENT_STATUS' signal and update minimum grade
+    requirement status.
+
+    Args:
+        sender: None
+        username(string): user name
+        grade_summary(dict): Dict containing output from the course grader
+        course_key(CourseKey): The key for the course
+        deadline(datetime): Course end date or None
+
+    Kwargs:
+        kwargs : None
+
+    """
+    from openedx.core.djangoapps.credit.api import (
+        is_credit_course, get_credit_requirement, set_credit_requirement_status
+    )
+
+    course_key = CourseKey.from_string(unicode(course_key))
```

Comment:
Since you're using `send_robust` and info-level logging, it's probably worth wrapping this in a general try/except block and logging at error-level if an exception occurs (unless Django is already doing this).